CONTRIBUTING.md

```diff
@@ -0,0 +1,36 @@
+# Contributing to the Astria monorepo
+
+Before an author submits a patch to the Astria monorepo at
+<https://github.com/astriaorg/astria>, they should make sure to take into
+account the guidelines laid out in this document.
+
+## How Pull Requests are written at Astria
+
+The following points represent an ideal scenario that patch authors should try
+to follow when submitting a patch in the form of a GitHub pull request (PRs):
+
+0. Be kind and respectful of your readers. Code is written once and read many
+   times. Be mindful that a good review can take as much or more time than
+   writing code.
+1. The pull request title follows
+   [conventional commits](https://www.conventionalcommits.org/en/v1.0.0/).
+2. A patch should be written in such a way that it is easy for a reader to
+   understand and review.
+3. Prefer shorter over longer patches.
+4. A patch should accomplish one goal, and accomplish it well.
+5. If a feature requires the codebase be refactored, refactor first, implement
+   the feature in a follow-up PR. The same applies to bugfixes.
+6. When moving around code in refactors (for example, moving a test module into
+   a separate file), do not also change item names or add new items. Remember
+   that GitHub diffs do not help a reviewer track which code is new, deleted,
+   or just moved.
+7. PR title and summary should give a concise and accurate representation of
+   what the patch wants to accomplish. They should not make assumptions about a
+   reader knowing implementation details. Ideally, put these in doc comments
+   that live next to the code.
+8. If a code change does not help accomplish the patch's goal described in 7,
+   it should be moved to another patch.
+9. Code should not be surprising - it should be idiomatic and fit in with the
+   rest of the code surrounding it. If a code changes is still necessary, flag
+   it for the reader using an inline comment, or adding a github review comment
+   pulling attention to it.
```

Comment:
I'm not quite following the last sentence.  Do you mean if further changes are still necessary, the author should flag these?

Seems like that could be a separate point on its own, and be more general, i.e. the author should flag any points of special interest that way to help out the reviewer?  (I really liked that approach when I first saw it here btw.)